pkg/clustermanager/cluster_creator_test.go

```diff
@@ -0,0 +1,75 @@
+package clustermanager_test
+
+import (
+	"context"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/pointer"
+
+	"github.com/aws/eks-anywhere/internal/test"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/clustermanager"
+	"github.com/aws/eks-anywhere/pkg/clustermanager/mocks"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	mockswriter "github.com/aws/eks-anywhere/pkg/filewriter/mocks"
+	"github.com/aws/eks-anywhere/pkg/kubeconfig"
+	mockskubeconfig "github.com/aws/eks-anywhere/pkg/kubeconfig/mocks"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+type clusterCreatorTest struct {
+	*WithT
+	ctx              context.Context
+	spec             *cluster.Spec
+	mgmtCluster      *types.Cluster
+	applier          *mocks.MockClusterApplier
+	writer           *mockswriter.MockFileWriter
+	kubeconfigWriter *mockskubeconfig.MockWriter
+}
+
+func newClusterCreator(t *testing.T) (*clustermanager.ClusterCreator, *clusterCreatorTest) {
+	ctrl := gomock.NewController(t)
+	cct := &clusterCreatorTest{
+		WithT:            NewWithT(t),
+		applier:          mocks.NewMockClusterApplier(ctrl),
+		writer:           mockswriter.NewMockFileWriter(ctrl),
+		kubeconfigWriter: mockskubeconfig.NewMockWriter(ctrl),
+		spec:             test.NewClusterSpec(),
+		ctx:              context.Background(),
+		mgmtCluster: &types.Cluster{
+			KubeconfigFile: "my-config",
+		},
+	}
+
+	cc := clustermanager.NewClusterCreator(cct.applier, cct.kubeconfigWriter, cct.writer)
+
+	return cc, cct
+}
+
+func (cct *clusterCreatorTest) expectFileCreate(clusterName, path string, w io.WriteCloser) {
+	cct.writer.EXPECT().Create(kubeconfig.FormatWorkloadClusterKubeconfigFilename(clusterName), gomock.AssignableToTypeOf([]filewriter.FileOptionsFunc{})).Return(w, path, nil)
```

Comment:
nit: i think it's better to hardcode here the value that you expect, considering that we control the cluster name, we should be able to do so